File: src/WinForms/coIT.BewirbDich.Winforms.Domain/Dokument.cs
namespace coIT.BewirbDich.Winforms.Domain;

public class Dokument
{
    public Guid Id { get; }

    public Dokumenttyp Typ { get; set; }
    public Berechnungsart Berechnungsart { get; set; }
    /// <summary>
    /// Berechnungsart Umsatz => Jahresumsatz in Euro,
    /// Berechnungsart Haushaltssumme => Haushaltssumme in Euro,
    /// Berechnungsart AnzahlMitarbeiter => Ganzzahl
    /// </summary>
    public decimal Berechnungbasis { get; set; }

    public bool InkludiereZusatzschutz { get; set; }
    public float ZusatzschutzAufschlag { get; set; }
    
    //Gibt es nur bei Unternehmen, die nach Umsatz abgerechnet werden
    public bool HatWebshop { get; set; }

    public Risiko Risiko { get; set; }
    
    public decimal Beitrag { get; set; }

    public bool VersicherungsscheinAusgestellt { get; set; }
    public decimal Versicherungssumme { get; set; }

    public Dokument()
    {
        Id = Guid.NewGuid();
    }
}

File: src/fullstack-angular-dotnet/creepy-client/src/app/models/dokument.ts
export interface DokumentenlisteEintragDto {
  id: string;
  dokumenttyp: string;
  berechnungsart: string;
  zusatzschutz: string
  webshopVersichert: boolean;
  risiko: string;
  versicherungssumme: number;
  beitrag: number;
  kannAngenommenWerden: boolean;
  kannAusgestelltWerden: boolean;
}

export interface ErzeugeNeuesAngebotDto {
  berechnungsart: string;
  willZusatzschutz: boolean;
  zusatzschutzAufschlag: string;
  hatWebshop: boolean;
  risiko: string;
  versicherungssumme: number;
}
